File: src/main/java/ua/softgroup/matrix/server/service/ProjectService.java
package ua.softgroup.matrix.server.service;

import ua.softgroup.matrix.api.model.datamodels.ProjectModel;
import ua.softgroup.matrix.api.model.datamodels.TimeModel;
import ua.softgroup.matrix.server.persistent.entity.Project;
import ua.softgroup.matrix.server.persistent.entity.User;

import java.util.Optional;
import java.util.Set;

public interface ProjectService extends GeneralEntityService<Project> {

    TimeModel saveStartWorkTime(String userToken, Long projectId);

    TimeModel saveEndWorkTime(String userToken, Long projectId);

    TimeModel saveCheckpointTime(String userToken, Long projectId, Integer idleTime);

    Set<ProjectModel> getUserActiveProjects(String token);

    Optional<Project> getBySupervisorIdAndUser(Long supervisorId, User user);

}

File: src/main/scala/ua/softgroup/matrix/server/service/ProjectService.scala
package ua.softgroup.matrix.server.service

import java.util
import java.util.Optional

import ua.softgroup.matrix.api.model.datamodels.{ProjectModel, TimeModel}
import ua.softgroup.matrix.server.persistent.entity.{Project, User}

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait ProjectService extends GenericEntityService[Project] {

  def saveStartWorkTime(userToken: String, projectId: Long): TimeModel

  def saveEndWorkTime(userToken: String, projectId: Long): TimeModel

  def saveCheckpointTime(userToken: String, projectId: Long, idleTime: Int): TimeModel

  def getUserActiveProjects(token: String): util.Set[ProjectModel]

  def getUserActiveProjects(userId: Long): Set[Project]

  def getBySupervisorIdAndUser(supervisorId: Long, user: User): Optional[Project]

  def getBySupervisorId(supervisorId: Long): Set[Project]

}
